src/test/java/org/sagebionetworks/bridge/services/AccountServiceTest.java

```diff
@@ -432,7 +433,41 @@ public void updateAccountSucceedsForStudyDesignerUpdatingTestAccount() throws Ex
     }
     
     @Test
-    public void updateAccountNotFound() {
+    public void updateAccountSucceedsForOrgAdminUpdatingAdminAccount() throws Exception {
+        RequestContext.set(new RequestContext.Builder()
+                .withCallerUserId("id")
+                // .withOrgSponsoredStudies(ImmutableSet.of(TEST_STUDY_ID))
+                .withCallerRoles(ImmutableSet.of(ORG_ADMIN)).build());
+        
+        Account account = mockGetAccountById(ACCOUNT_ID, false);
+        account.setRoles(ImmutableSet.of(STUDY_DESIGNER));
+        // account.setDataGroups(ImmutableSet.of(TEST_USER_GROUP));
+        // account.setEnrollments(ImmutableSet.of(Enrollment.create(TEST_APP_ID, TEST_STUDY_ID, TEST_USER_ID)));
+
+        service.updateAccount(account);
+        
+        verify(mockAccountDao).updateAccount(account);
+    }
+    
+    @Test
+    public void updateAccountFailsForOrgAdminUpdatingParticipantAccount() throws Exception {
+        RequestContext.set(new RequestContext.Builder()
+                .withCallerUserId("id")
+                .withOrgSponsoredStudies(ImmutableSet.of(TEST_STUDY_ID))
+                .withCallerRoles(ImmutableSet.of(ORG_ADMIN)).build());
+        
+        Account account = mockGetAccountById(ACCOUNT_ID, false);
+        // account.setRoles(ImmutableSet.of(STUDY_DESIGNER));
+        account.setDataGroups(ImmutableSet.of(TEST_USER_GROUP));
+        account.setEnrollments(ImmutableSet.of(Enrollment.create(TEST_APP_ID, TEST_STUDY_ID, TEST_USER_ID)));
+
+        service.updateAccount(account);
+        
+        verify(mockAccountDao).updateAccount(account);
```

Comment:
This succeeds, this doesn't fail. Is this correct?

Reply:
Yep the service needed to be changed. It is changed. Long-term as mentioned, the solution is to have an AccountService for admin accounts and a ParticipantService for participants, and not allow them to cross.